THRIFT-2827: C++ - Remove unused tr1/functional include from ProcessorTest

Client: C++
    Patch: Aiden Scandella

This closes #267

Include was unused yet was causing compilation problems on platforms
        without c++11 tr1

diff --git a/lib/cpp/test/processor/ProcessorTest.cpp b/lib/cpp/test/processor/ProcessorTest.cpp
index 8b806b4..eee90c7 100644
--- a/lib/cpp/test/processor/ProcessorTest.cpp
+++ b/lib/cpp/test/processor/ProcessorTest.cpp
@@ -23,7 +23,6 @@
  * implementations.
  */
 
-#include <tr1/functional>
 #include <boost/test/unit_test.hpp>
 
 #include <thrift/concurrency/PosixThreadFactory.h>
